lib/server-impl.js

```diff
@@ -26,13 +26,17 @@ function createApp(options) {
     );
 
     const app = getApp(config);
-    const server = app.listen(app.get('port'), () => {
-        logger.info(`Unleash started on ${app.get('port')}`);
-    });
-
     startMonitoring(options.serverMetrics, eventBus);
 
-    return { app, server, eventBus };
+    return new Promise(resolve => {
+        const server = app.listen(
+            { port: options.port, host: options.host },
+            () => {
+                logger.info(`Unleash started on port ${server.address().port}`);
+                resolve({ app, server, eventBus });
+            }
+        );
+    });
```

Comment:
should we do `server.on('error', reject)` in here?

Reply:
could that lead to a resolved promise being rejected?

Reply:
using `listening` and `error` events to complete the promise should be safe.